onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -2674,6 +2672,282 @@ Status LRNOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const No
+  if (node.InputDefs().size() > 1)
+    model_builder.AddInitializerToSkip(node.InputDefs()[1]->Name());  // min
+
+  if (node.InputDefs().size() > 2)
+    model_builder.AddInitializerToSkip(node.InputDefs()[2]->Name());  // max
+}
+
+/* static */ bool ClipOpBuilder::GetMinMax(ModelBuilder& model_builder, const Node& node, float& min, float& max) {
+  const auto& initializers(model_builder.GetInitializerTensors());
+
+  if (node.InputDefs().size() > 1) {  // we have input min
+    const auto& min_name = node.InputDefs()[1]->Name();
+    if (!Contains(initializers, min_name)) {
+      LOGS_DEFAULT(VERBOSE) << "Input min of Clip must be known";
+      return false;
+    }
+    min = GetTensorFloatData(initializers.at(min_name))[0];
+  }
+
+  if (node.InputDefs().size() > 2) {  // we have input max
+    const auto& max_name = node.InputDefs()[2]->Name();
+    if (!Contains(initializers, max_name)) {
+      LOGS_DEFAULT(VERBOSE) << "Input max of Clip must be known";
+      return false;
+    }
+    max = GetTensorFloatData(initializers.at(max_name))[0];
+  }
+
+  return true;
+}
+
+bool ClipOpBuilder::IsOpSupportedImpl(ModelBuilder& model_builder, const Node& node) {
+  if (node.SinceVersion() < 11) {
+    LOGS_DEFAULT(VERBOSE) << "Clip only supports opset 11+";
+    return false;
+  }
+
+  float min = std::numeric_limits<float>::lowest();
+  float max = std::numeric_limits<float>::max();
+  if (!GetMinMax(model_builder, node, min, max))
+    return false;
+
+  // We only supoort relu6 or relu1
+  // TODO, support clip between 2 arbitrary numbers
+  if ((min == 0.0f && max == 6.0f) || (min == -1.0f && max == 1.0f)) {
+    return true;
+  } else {
+    LOGS_DEFAULT(VERBOSE) << "Clip only supports [min, max] = [0, 6] or [-1, 1], the input is ["
+                          << min << ", " << max << "]";
+    return false;
+  }
+
+  return true;
+}
+
+Status ClipOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node) {
+  auto& shaper(model_builder.GetShaper());
+  const auto& operand_indices(model_builder.GetOperandIndices());
+  const auto& operand_types(model_builder.GetOperandTypes());
+
+  const auto& input = node.InputDefs()[0]->Name();
+  const auto& output = node.OutputDefs()[0]->Name();
+  bool output_is_nhwc = model_builder.IsOperandNHWC(input);
+
+  ORT_RETURN_IF_ERROR(shaper.Identity(input, output));
+  const OperandType output_operand_type(operand_types.at(input).type, shaper[output]);
+
+  float min = std::numeric_limits<float>::lowest();
+  float max = std::numeric_limits<float>::max();
+  GetMinMax(model_builder, node, min, max);
+
+  int32_t op_code;
+  if (min == 0.0f && max == 6.0f)
+    op_code = ANEURALNETWORKS_RELU6;
+  else if (min == -1.0f && max == 1.0f)
+    op_code = ANEURALNETWORKS_RELU1;
+  else
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "ClipOpBuilder, unsupported input [", min, ", ", max, "]");
+
```

Comment:
If this gets hit is it because IsOpSupportedImpl doesn't match the logic here? If so the error message may want to call that out.